extensions/markdown-language-features/src/features/smartSelect.ts

```diff
@@ -123,6 +127,82 @@ function createBlockRange(block: Token, document: vscode.TextDocument, cursorLin
 	}
 }
 
+function createInlineRange(document: vscode.TextDocument, cursorPosition: vscode.Position, parent?: vscode.SelectionRange): vscode.SelectionRange | undefined {
+	const line = document.lineAt(cursorPosition.line).text;
+
+	// find closest ** that occurs before cursor position
+	let startBold = line.substring(0, cursorPosition.character).lastIndexOf('**');
+
+	// find closest ** that occurs after the start **
+	const endBoldIndex = line.substring(startBold + 2).indexOf('**');
+	let endBold = startBold + 2 + line.substring(startBold + 2).indexOf('**');
+
+	if (startBold >= 0 && endBoldIndex >= 0 && startBold + 1 < endBold && startBold <= cursorPosition.character && endBold >= cursorPosition.character) {
+		const range = new vscode.Range(cursorPosition.line, startBold, cursorPosition.line, endBold + 2);
+		// **content cursor content** so select content then ** on both sides
+		const contentRange = new vscode.Range(cursorPosition.line, startBold + 2, cursorPosition.line, endBold);
+		return new vscode.SelectionRange(contentRange, new vscode.SelectionRange(range, parent));
+	} else if (startBold >= 0) {
+		// **content**cursor or **content*cursor*
+		// find end ** from end of start ** to end of line (since the cursor is within the end stars)
+		let adjustedEnd = startBold + 2 + line.substring(startBold + 2).indexOf('**');
+		startBold = line.substring(0, adjustedEnd - 2).lastIndexOf('**');
+		if (adjustedEnd >= 0 && cursorPosition.character === adjustedEnd || cursorPosition.character === adjustedEnd + 1) {
+			if (document.lineAt(cursorPosition.line).text.charAt(adjustedEnd + 1) === '*') {
+				// *cursor* so need to extend end to include the second *
+				adjustedEnd += 1;
+			}
+			return new vscode.SelectionRange(new vscode.Range(cursorPosition.line, startBold, cursorPosition.line, adjustedEnd + 1), parent);
+		}
+	} else if (endBold > 0) {
+		// cursor**content** or *cursor*content**
+		// find start ** from start of string to cursor + 2 (since the cursor is within the start stars)
+		const adjustedStart = line.substring(0, cursorPosition.character + 2).lastIndexOf('**');
+		endBold = adjustedStart + 2 + line.substring(adjustedStart + 2).indexOf('**');
+		if (adjustedStart >= 0 && adjustedStart === cursorPosition.character || adjustedStart === cursorPosition.character - 1) {
+			return new vscode.SelectionRange(new vscode.Range(cursorPosition.line, adjustedStart, cursorPosition.line, endBold + 2), parent);
+		}
+	}
+
+	const startBracket = line.substring(0, cursorPosition.character + 1).lastIndexOf('[');
+	let endBracket = line.substring(cursorPosition.character).indexOf(']');
+
+	if (startBracket >= 0 && endBracket >= 0) {
```

Comment:
do we want selection of endbracket and endparens ?

for example, 

(word)| where | is the cursor? I thought no because in cases such as 

`[request](https://github.com/microsoft/vscode/issues/56704)` , we want to be able to select the whole link with the parens with cursor here `|(https://github.com/microsoft/vscode/issues/56704)`